src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -91,16 +92,15 @@ public class TimetableSnapshot {
    * The compound key approach better reflects the fact that there should be only one Timetable per
    * TripPattern and date.
    */
-  private HashMap<TripPattern, SortedSet<Timetable>> timetables = new HashMap();
+  private Map<TripPattern, SortedSet<Timetable>> timetables = new HashMap<>();
```

Comment:
I wonder if it would make sense to convert this to a [`SortedSetMultimap`](https://guava.dev/releases/23.0/api/docs/com/google/common/collect/SortedSetMultimap.html).

Reply:
That would make the intent more obvious, but at the same time that would hide the SortedSet as an implementation detail of the MultiMap. But we need to keep control of this SortedSet to be able to implement copy-and-write (and to replace it by an ImmutableSortedSet) as done in #5941